### PR TITLE
Start and end handlers for iframe login

### PR DESCRIPTION
**Pull request by nathanstitt**
This allows Tutor to set the return point once login completes and then relay the access token back to the iframe once the completion page is loaded

### CODE REVIEW

File: app/controllers/auth_controller.rb [hunk 1]
```diff
@@ -1,20 +1,21 @@
...
+  before_filter :allow_iframe_access, only: [:iframe_start, :iframe_finish]
+
+  #before_filter :forbid_access
+  # these should always return 200 response regardless of login status
+  skip_before_filter :authenticate_user!, only: [:status, :cors_preflight_check, :iframe_start, :iframe_finish]
+
+
+
+  # Since these endpoints are loaded from foreign sites via cors or iframe, CRSF tokens can't be used
+  skip_before_action :verify_authenticity_token
```
Comment (jpslav) [inline app/controllers/auth_controller.rb]: Can you explicitly add the `only: blah` here -- I like to skip explicitly so others don't get burned down the line if they add a new route.  You can also do

IFRAME_OR_CORS_ACTIONS = [:status, :cors_preflight_check, :iframe_start, :iframe_finish]
skip_before_filter :authenticate_user!, only: IFRAME_OR_CORS_ACTIONS
etc

Comment (nathanstitt) [inline app/controllers/auth_controller.rb]: Yep, I just added them explicitly.  There's only two that use the complete list so isn't worth adding a new constant for them quite yet (imo).

**Comment by jpslav** [comment]
Thanks @nathanstitt, feel free to merge when CI passes.

**Comment by nathanstitt** [comment]
Merging.  They'll be a follow up shortly to display T&C,  and stub account choosing for dev

**Pull request merged**